app/bundles/EmailBundle/Form/Type/DashboardEmailsInTimeWidgetType.php

```diff
@@ -40,6 +77,49 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'required'   => false,
             ]
         );
+        $companies        = $this->companyRepository->getCompanies();
+        $companiesChoises = [];
+        foreach ($companies as $company) {
+            $companiesChoises[$company['id']] = $company['companyname'];
+        }
+        $builder->add('companyId', 'choice', [
+            'label'      => 'mautic.email.companyId.filter',
+            'choices'    => $companiesChoises,
+            'label_attr' => ['class' => 'control-label'],
+            'attr'       => ['class' => 'form-control'],
+            'empty_data' => '',
+            'required'   => false,
+        ]);
+        /** @var Campaign[] $campaigns */
+        $campaigns        = $this->campaignRepository->findAll();
+        $campaignsChoices = [];
+        foreach ($campaigns as $campaign) {
+            $campaignsChoices[$campaign->getId()] = $campaign->getName();
+        }
+        $builder->add('campaignId', 'choice', [
+                'label'      => 'mautic.email.campaignId.filter',
+                'choices'    => $campaignsChoices,
+                'label_attr' => ['class' => 'control-label'],
+                'attr'       => ['class' => 'form-control'],
+                'empty_data' => '',
+                'required'   => false,
+            ]
+        );
+        /** @var LeadList[] $segments */
+        $segments        = $this->segmentsRepository->findAll();
```

Comment:
This will load all segments. Use `LeadListRepository::getLists($user)` to get only segments that the user can see. The same for the list of companies and campaigns.